Make Project XC8 and CCI compliant

taylortrimble opened the pull request:
This is a work in progress. I'd like to define some clean, consistent coding style here and also make our project use the super-sexy new (aka non-deprecated) XC8 compiler with CCI portability.

**Use this Pull Request to discuss coding style as well.**

We'll have to manually merge most of our code at one point or another because of this, but @aterlumen  or I can help with that.

Commit:
Some messiness to unify comment and section style

Commit:
Factor out I2C functions

Commit:
Make projects CCI compliant

Move global variables for I2C to `I2C.c` file

Commit:
Use CCI line endings

UTF-8 LF only

CrasherCourse commented on masterChipSendSignal.c:
```diff
@@ -1,158 +1,97 @@
+// Interrupts
+//
+
+__interrupt(high_priority) void getData(void)
```
I getting some compiler errors here.  Are we using "--STRICT" option?  That's the only time we use _interrupts.
For me to compile this I needed to use:
void interrupt high_priority getData(void)

Also, wouldn't be better to give getData its own c and header files since it handles the DMX interpretation?
Or at least put the bulk of the code into functions in another file?

taylortrimble replied:
Ah! Yes, you need to set your XC8 compiler to "Use CCI Syntax". If you right-click your project and go to "Properties", you can set the "XC8 global options -> XC8 Compiler -> Use CCI Syntax" check box.

I'll add a note to the README, and rename the interrupt - it wasn't supposed to be `getData`, it was just copied and pasted from an example!

Commit:
Rename confusing Hi-Pri interrupt

Use consistent function naming convention (smalllettersruntogether for
system calls, PascalCase for our defined functions, _PascalCase for
"private" functions)

Commit:
Tabs to spaces

Commit:
Fix for loop indentation

Commit:
Add instructions for compiling with CCI syntax enabled

taylortrimble commented:
@CrasherCourse @aterlumen EEW! I'm old and crusty and nobody's said anything yet! Am I ready to pull?

CrasherCourse commented:
I looked through the code, didn't find any major errors, and it compiles just fine.
I do get advisory 1233 and 1234 and warning 1273 when I compile in the old MPLAB.
1234 is a little scary since it's says corrupted data can occur (if async reset occurs).

A small note:
The predefined values of slave address and the first register of the buffer were just for seeing if the debugger would let's us test I2C at the early stages of the code.  We never got it to work back then.  You can delete those values if you want.

Otherwise, I fairly confident that you can merge these files.

kevana commented:
At band til 11, can't fix anything until then, can probably merge after the
formatting stuff is fixed.
On Nov 7, 2012 6:24 PM, "Taylor Trimble" notifications@github.com wrote:

> @CrasherCourse https://github.com/CrasherCourse @aterlumenhttps://github.com/aterlumenEEW! I'm old and crusty and nobody's said anything yet! Am I ready to pull?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/teslaworksumn/HeadMaster/pull/14#issuecomment-10171290.

taylortrimble commented:
@aterlumen Yeah I'll probably wait until #15 is in, fart with it some more, and then merge this in.
@CrasherCourse Thanks for the thorough testing! Nice :clap: I'll get rid of the testing stuff in a future Pull Request, I'll keep this one laser focused on compiler though. Opening an issue preemptively.

Commit:
Merge changes from master

taylortrimble commented:
This is complex and will never be able to be auto-merged: FYI

taylortrimble commented:
Okay, Pulling #15 left some build errors, which is okay! We'll fix those here as we transition to the new compiler, which may be lenient on the stuff that is causing #15 to fail. I'll add more throughout the day, **please check it out** so we can get this in ASAP.

Thanks!

Commit:
Correct function name

Case matters!

Commit:
Include mandatory xc.h

Commit:
Use defines for integer constants

This enables us to use constants for defining the size of statically
allocated arrays

taylortrimble commented:
@CrasherCourse @aterlumen Hey guys! Review this please! I think it could be ready to pull soon :wink:

CrasherCourse commented:
Tested the branch and it compiles for me.

taylortrimble commented:
:grinning:

kevana commented:
Go for it.

taylortrimble commented:
Woot! Thanks for all the help with this guys!

John-Phillips commented:
AAAAAAAAAAAAAAAhhhhhhhhhhh yea @aterlumen @CrasherCourse and @tylrtrmbl! You all are doing fantastic work. Keep on rocking!
